src/components/pages/CompanySubscriptions/CompanySubscriptionDetail.tsx

```diff
@@ -17,37 +17,75 @@
   useFetchAppDetailsQuery,
   useFetchSubscriptionAppQuery,
 } from 'features/apps/apiSlice'
 import { PAGES } from 'types/Constants'
+import './CompanySubscriptions.scss'
+import {
+  useFetchServiceDetailsQuery,
+  useFetchSubscriptionServiceQuery,
+} from 'features/serviceSubscription/serviceSubscriptionApiSlice'
+import CommonService from 'services/CommonService'
+import { useState, useEffect } from 'react'
+import { getApiBase } from 'services/EnvironmentService'
 import CompanySubscriptionTechnical from './components/CompanySubscriptionTechnical'
-import CompanySubscriptionContent from './components/CompanySubscriptionContent'
 import CompanySubscriptionHeader from './components/CompanySubscriptionHeader'
-import CompanySubscriptionDocument from './components/CompanySubscriptionDocument'
+import CompanySubscriptionContent from './components/CompanySubscriptionContent'
 import CompanySubscriptionPrivacy from './components/CompanySubscriptionPrivacyContent'
-import './CompanySubscriptions.scss'
+import CompanySubscriptionDocument from './components/CompanySubscriptionDocument'
 
 export default function CompanySubscriptionDetail() {
   const navigate = useNavigate()
   const { state: items } = useLocation()
   const { t } = useTranslation()
+  const id = items.row.offerId ?? ('' as string)
+  const subscriptionId = items.row.subscriptionId ?? ('' as string)
+  const { data: appData, error: appError } = useFetchSubscriptionAppQuery(
+    { appId: id, subscriptionId },
+    { skip: items.service }
+  )
+  const { data: serviceData, error: serviceError } =
+    useFetchSubscriptionServiceQuery(
+      { serviceId: id, subscriptionId },
+      { skip: items.app }
+    )
+  const { data: fetchAppsData } = useFetchAppDetailsQuery(id, {
+    skip: items.service,
+  })
+  const { data: fetchServicessData } = useFetchServiceDetailsQuery(id, {
+    skip: items.app,
+  })
+  const [docId, setDocId] = useState('')
+
+  const data = items.app ? appData : serviceData
+  const fetchData = items.app ? fetchAppsData : fetchServicessData
 
-  const appId = items ? items.offerId : ''
-  const subscriptionId = items ? items.subscriptionId : ''
+  // To-Do fix the type issue with status and data from FetchBaseQueryError
+  // eslint-disable-next-line
+  const error: any = items.app ? appError : serviceError
 
-  // Prevent API call when appId does not exist
-  const { data } = appId
-    ? useFetchSubscriptionAppQuery({ appId, subscriptionId })
-    : { data: undefined }
-  const { data: fetchAppsData } = appId
-    ? useFetchAppDetailsQuery(appId)
-    : { data: undefined }
+  useEffect(() => {
+    if (fetchData?.leadPictureId) {
+      const id = CommonService.isValidPictureId(fetchData?.leadPictureId)
+      setDocId(id)
+    }
+  }, [fetchData])
+
+  const getSrc = () => {
+    if (fetchData?.id && items.app && docId)
+      return `${getApiBase()}/api/apps/${fetchData.id}/appDocuments/${docId}`
+    else if (fetchData?.id && items.service && docId)
```

Comment:
`else` not needed because return before

Reply:
currently it is,
```
if()
return
else if()
return
return
```
you mean this??
```
if()
return
if()
return
return
```

Reply:
yes, because in the case the if block returns guaranteed then we don't need the else and can simply continue with another if because it's obvious that's the else case.